OpenSim/Simulation/Model/Muscle.cpp

```diff
@@ -120,6 +121,19 @@ void Muscle::updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber)
                 }
             }
         }
+        if (versionNumber < 30516) {
+            // Find GeometryPath node and insert <default_color>
+            SimTK::Xml::element_iterator  geomPathIter = aNode.element_begin("GeometryPath");
+            if (geomPathIter != aNode.element_end()) {
+                SimTK::Xml::element_iterator  defaultColorIter = geomPathIter->element_begin("default_color");
+                if (defaultColorIter == geomPathIter->element_end()) {
+                    SimTK::Xml::Element myDefaultColorEement("default_color");
+                    myDefaultColorEement.setValue(".8 .1 .1"); // DefaultMuscleColor
+                    geomPathIter->appendNode(myDefaultColorEement);
+                }
+            }
+        }
```

Comment:
Deserialized objects don't go through the constructProperties() call as far as I could tell (by stepping through the code) that's why I use the updateFromXMLNode route. In fact if I remove this block the red color was never set on deserialization. Newer versions will serialize with a color and will not go through this block of code (as verified by testing round-trip serialize/deserialize models with arbitrary colored muscles)